### PR TITLE
added LV,FS and Software raid to dbench,fs_mark,iozone,ltp_fsstress

### PR DESCRIPTION
**Pull request by smruti77**
Test creates FS, LV software raid and runs test on them, as an option.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[dbench.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4970365/dbench.log)
[fs_mark.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4970370/fs_mark.log)
[iozone.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4970371/iozone.log)

**Comment by smruti77** [comment]
JOB ID     : 3fa2283c1a8b6b365fdd5fa6ed7580dc07a3c4f7
JOB LOG    : /root/avocado/job-results/job-2020-07-22T03.50-3fa2283/job.log
 (01/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext4-lv-raid-b6e4: PASS (218.15 s)
 (02/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext4-lv-no_raid-82cf: PASS (298.96 s)
 (03/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext4-no_lv-raid-3a74: PASS (295.99 s)
 (04/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext4-no_lv-no_raid-c9c3: PASS (294.14 s)
 (05/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext3-lv-raid-cd8f: PASS (294.91 s)
 (06/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext3-lv-no_raid-6077: PASS (300.75 s)
 (07/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext3-no_lv-raid-3b6c: PASS (295.49 s)
 (08/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext3-no_lv-no_raid-a27d: PASS (294.36 s)
 (09/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext2-lv-raid-d87f: PASS (298.09 s)
 (10/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext2-lv-no_raid-0edf: PASS (300.22 s)
 (11/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext2-no_lv-raid-af7c: PASS (296.89 s)
 (12/24) ltp_fsstress.py:LtpFs.test_fsstress_run;ext2-no_lv-no_raid-dd9f: PASS (296.24 s)
 (13/24) ltp_fsstress.py:LtpFs.test_fsstress_run;xfs-lv-raid-7bb8: PASS (298.19 s)
 (14/24) ltp_fsstress.py:LtpFs.test_fsstress_run;xfs-lv-no_raid-4d6e: PASS (300.99 s)
 (15/24) ltp_fsstress.py:LtpFs.test_fsstress_run;xfs-no_lv-raid-9eb9: PASS (293.57 s)
 (16/24) ltp_fsstress.py:LtpFs.test_fsstress_run;xfs-no_lv-no_raid-0112: PASS (296.44 s)
 (17/24) ltp_fsstress.py:LtpFs.test_fsstress_run;btrfs-lv-raid-bf7f: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.50 s)
 (18/24) ltp_fsstress.py:LtpFs.test_fsstress_run;btrfs-lv-no_raid-c478: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.49 s)
 (19/24) ltp_fsstress.py:LtpFs.test_fsstress_run;btrfs-no_lv-raid-0c82: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.50 s)
 (20/24) ltp_fsstress.py:LtpFs.test_fsstress_run;btrfs-no_lv-no_raid-d81f: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.49 s)
 (21/24) ltp_fsstress.py:LtpFs.test_fsstress_run;no_fs-lv-raid-17ab: PASS (300.14 s)
 (22/24) ltp_fsstress.py:LtpFs.test_fsstress_run;no_fs-lv-no_raid-1068: PASS (299.14 s)
 (23/24) ltp_fsstress.py:LtpFs.test_fsstress_run;no_fs-no_lv-raid-68ca: PASS (294.50 s)
 (24/24) ltp_fsstress.py:LtpFs.test_fsstress_run;no_fs-no_lv-no_raid-038a: PASS (294.39 s)
RESULTS    : PASS 20 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB HTML   : /root/avocado/job-results/job-2020-07-22T03.50-3fa2283/results.html
JOB TIME   : 5864.08 s

JOB ID     : 57b9d0483adefb4dff30b436dcc1613d1a521ff6
JOB LOG    : /root/avocado/job-results/job-2020-07-24T00.51-57b9d04/job.log
 (01/16) fs_mark.py:FSMark.test;dir-ext4-lv-raid-d6e5: PASS (8.17 s)
 (02/16) fs_mark.py:FSMark.test;dir-ext4-lv-no_raid-f11c: PASS (7.07 s)
 (03/16) fs_mark.py:FSMark.test;dir-ext4-no_lv-raid-c0fe: PASS (4.77 s)
 (04/16) fs_mark.py:FSMark.test;dir-ext4-no_lv-no_raid-d0ef: PASS (4.14 s)
 (05/16) fs_mark.py:FSMark.test;dir-xfs-lv-raid-9597: PASS (6.15 s)
 (06/16) fs_mark.py:FSMark.test;dir-xfs-lv-no_raid-d8d1: PASS (6.29 s)
 (07/16) fs_mark.py:FSMark.test;dir-xfs-no_lv-raid-59fb: PASS (4.25 s)
 (08/16) fs_mark.py:FSMark.test;dir-xfs-no_lv-no_raid-c76b: PASS (3.29 s)
 (09/16) fs_mark.py:FSMark.test;dir-btrfs-lv-raid-9485: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.96 s)
 (10/16) fs_mark.py:FSMark.test;dir-btrfs-lv-no_raid-15ca: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.40 s)
 (11/16) fs_mark.py:FSMark.test;dir-btrfs-no_lv-raid-c958: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.96 s)
 (12/16) fs_mark.py:FSMark.test;dir-btrfs-no_lv-no_raid-f2b8: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.40 s)
 (13/16) fs_mark.py:FSMark.test;dir-no_fs-lv-raid-d150: ERROR: Command './fs_mark -d /tmp/link -s 10240 -n 1000' failed.\nstdout: b'\n#  ./fs_mark  -d  /tmp/link  -s  10240  -n  1000 \n#\tVersion 3.3, 1 thread(s) starting at Fri Jul 24 00:52:25 2020\n#\tSync method: INBAND FSYNC: fsync() per file in write loop.\n#\tDi... (3.69 s)
 (14/16) fs_mark.py:FSMark.test;dir-no_fs-lv-no_raid-67ac: ERROR: Command './fs_mark -d /tmp/link -s 10240 -n 1000' failed.\nstdout: b'\n#  ./fs_mark  -d  /tmp/link  -s  10240  -n  1000 \n#\tVersion 3.3, 1 thread(s) starting at Fri Jul 24 00:52:29 2020\n#\tSync method: INBAND FSYNC: fsync() per file in write loop.\n#\tDi... (3.81 s)
 (15/16) fs_mark.py:FSMark.test;dir-no_fs-no_lv-raid-2aa1: PASS (2.75 s)
 (16/16) fs_mark.py:FSMark.test;dir-no_fs-no_lv-no_raid-336b: PASS (1.62 s)
RESULTS    : PASS 10 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB HTML   : /root/avocado/job-results/job-2020-07-24T00.51-57b9d04/results.html

**Comment by smruti77** [comment]
[dbench.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5067521/dbench.log)
[fs_mark.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5067523/fs_mark.log)
[iozone.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5067524/iozone.log)
[ltp_fsstress.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5067525/ltp_fsstress.log)

File: io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml [hunk 1]
```diff
@@ -2,3 +2,26 @@ disk:
...
+    ext3:
+        fs: 'ext3'
+    ext2:
+        fs: 'ext2'
+    xfs:
+        fs: 'xfs'
+    btrfs:
+        fs: 'btrfs'
+    no_fs:
+        fs: ''
```
Comment (narasimhan-v) [inline io/disk/ltp_fsstress.py.data/ltp_fsstress.yaml:17]: Here also it is not supported.

File: io/disk/dbench.py.data/dbench.yaml [hunk 2]
```diff
@@ -12,4 +13,22 @@ setup:
     location:
         default:
             dir: '.'
```
Comment (narasimhan-v) [inline io/disk/dbench.py.data/dbench.yaml:15]: @smruti77 This should be removed.

Comment (narasimhan-v) [reply io/disk/dbench.py.data/dbench.yaml:15]: And the test should be run on `self.dirs`, not dir.

File: io/disk/iozone.py.data/iozone.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+disk:
 setup:
     location: !mux
         default:
```
Comment (narasimhan-v) [inline io/disk/iozone.py.data/iozone.yaml:4]: This also should be removed, and test should be run on `self.dirs`

File: io/disk/dbench.py.data/dbench.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+disk:
 setup:
     duration: !mux
         default:
```
Comment (narasimhan-v) [inline io/disk/dbench.py.data/dbench.yaml:4]: @smruti77 you need to remove these lines as well.
And make sure Readme is also updated, if needed, in all files.

File: io/disk/dbench.py.data/dbench.yaml [hunk 5]
```diff
@@ -11,5 +12,22 @@ setup:
             nprocs: 1
     location:
         default:
```
Comment (narasimhan-v) [inline io/disk/dbench.py.data/dbench.yaml:14]: @smruti77 you need to remove these lines as well.
And make sure Readme is also updated, if needed, in all files.

**Pull request merged**